### PR TITLE
Environments depends on recipe and model

### LINKED ISSUE #281: environmentManager should consider environment as (recipe, model), not only (recipe)

### PR DESCRIPTION
**Pull request by feloy**
### What does this PR do?

Index the environments (applications) by (recipeId, modelId) instead of recipeId only.

### What issues does this PR fix or reference?

Fixes #281 

### How to test this PR?

Start, stop, restart applications from environments page and recipe page. Should work as before

### CODE REVIEW

File: packages/shared/src/models/IRecipeModelIndex.ts [hunk 1]
```diff
@@ -0,0 +1,4 @@
+export interface RecipeModelIndex {
```
Comment (axel7083) [inline packages/shared/src/models/IRecipeModelIndex.ts:1]: What is the purpose of this interface ?

Comment (axel7083) [inline packages/shared/src/models/IRecipeModelIndex.ts:1]: copyrights ?

**Review by axel7083 — APPROVED** [review]
LGTM

File: packages/shared/src/models/IRecipeModelIndex.ts (hunk 1 above)
Comment (feloy) [reply packages/shared/src/models/IRecipeModelIndex.ts:1]: Represents the type of the key of the `ApplicationRegistry` class:

```
export class ApplicationRegistry<T extends RecipeModelIndex> {
```

**Pull request merged**